### PR TITLE
Add log message on verbose, when beets renames destination file during move.

### PR DESCRIPTION
**Pull request by fleur**
## Description

First step in fix of #2895.  Add log message on verbose, when beets renames destination file during move.  The issue says it may be trickier than it looks.  It did not seem tricky, so only doing the first step suggested, to see if I missed something.

## To Do

- [x] Documentation is not needed since functional behavior is not changing.
- [x] Changelog
- [x] Tests are not needed because there are already tests in test/test_files.py that exercise the code path that pass.

### CODE REVIEW

File: beets/library.py [hunk 1]
```diff
@@ -856,7 +856,13 @@ def move_file(self, dest, operation=MoveOperation.MOVE):
         `operation` should be an instance of `util.MoveOperation`.
         """
         if not util.samefile(self.path, dest):
+            old_dest = dest
             dest = util.unique_path(dest)
+            log.debug(
+                'File already exists at dest, changing final path slightly, '
+                "to be unique, from '{}', to '{}'.",
```
Comment (sampsyo) [inline beets/library.py:863]: Just a suggestion to slightly simplify the wording here… how about this?

> File exists at `{}`. Avoiding conflict with new name: `{}`

File: beets/library.py [hunk 2]
```diff
@@ -856,7 +856,13 @@ def move_file(self, dest, operation=MoveOperation.MOVE):
         `operation` should be an instance of `util.MoveOperation`.
         """
         if not util.samefile(self.path, dest):
+            old_dest = dest
             dest = util.unique_path(dest)
+            log.debug(
```
Comment (sampsyo) [inline beets/library.py:861]: I think you may need a `if dest != old_dest` here. This may also need some careful testing to make sure it doesn't fire spuriously when these point to the same path but are just normalized differently… hopefully that doesn't happen, but a few test runs should reveal the answer!

**Review by sampsyo — COMMENTED** [review]
This seems like a nice idea! Here are some suggestions.

File: beets/library.py [hunk 3]
```diff
@@ -856,7 +856,13 @@ def move_file(self, dest, operation=MoveOperation.MOVE):
...
+                'File already exists at dest, changing final path slightly, '
+                "to be unique, from '{}', to '{}'.",
+                old_dest, dest
```
Comment (sampsyo) [inline beets/library.py:864]: These will need to use `displayable_path` wrappers.

File: beets/library.py (hunk 2 above)
Comment (fleur) [reply beets/library.py:861]: If you have any specific suggestions of test runs, and have the time to, please let me know.  All tests pass, so I'm assuming you mean something outside of those.

Comment (sampsyo) [reply beets/library.py:861]: About being "normalized differently," I'm just thinking about all the various ways in which different paths can refer to the same file. For example, `os.path.normpath` or `os.path.abspath` are different ways of transforming paths so they still refer to the same file. It's just worth double-checking to make sure that we're not doing that kind of thing when transforming the path (or that the original path is already normalized in the same way, which it probably is).

I suggested `if dest != old_dest` so that (normalization aside) this message only gets printed when there is an actual change.

For testing, all I mean is running this on a few files—perhaps in the same cases that presumably motivated you to add this log message—and make sure that the message shows up when it's supposed to (and doesn't appear when it isn't).

**Comment by JOJ0** [comment]
Is there something to do here still @fleur? Was the `old != new` suggestion sorted out? Or new ideas? I think this is a candidate for merging soon :-) HTH
